# Manual prompting with ChatGPT

## Load data

In [14]:
import yfinance as yf

stocks = ['AAPL', 'MSFT', 'AMZN', 'GOOGL', 'GOOG', 'META', 'TSLA']
df_base = yf.download(stocks, start='2010-01-01', end='2020-12-31')

[*********************100%%**********************]  7 of 7 completed


In [15]:
df_base

Price        Adj Close                                                \
Ticker            AAPL        AMZN       GOOG      GOOGL        META   
Date                                                                   
2010-01-04    6.461976    6.695000  15.592496  15.666453         NaN   
2010-01-05    6.473148    6.734500  15.523831  15.597464         NaN   
...                ...         ...        ...        ...         ...   
2020-12-29  132.181183  166.100006  87.836044  87.787247  276.212402   
2020-12-30  131.054169  164.292496  86.877136  86.712982  271.312469   

Price                                    Close                         \
Ticker            MSFT        TSLA        AAPL        AMZN       GOOG   
Date                                                                    
2010-01-04   23.389406         NaN    7.643214    6.695000  15.610239   
2010-01-05   23.396950         NaN    7.656429    6.734500  15.541497   
...                ...         ...         ...         ...        ...   
2020-12-29  217.492432  221.996674  134.869995  166.100006  87.935997   
2020-12-30  215.095795  231.593338  133.720001  164.292496  86.975998   

Price                                                            High  \
Ticker          GOOGL        META        MSFT        TSLA        AAPL   
Date                                                                    
2010-01-04  15.684434         NaN   30.950001         NaN    7.660714   
2010-01-05  15.615365         NaN   30.959999         NaN    7.699643   
...               ...         ...         ...         ...         ...   
2020-12-29  87.888000  276.779999  224.149994  221.996674  138.789993   
2020-12-30  86.812500  271.869995  221.679993  231.593338  135.990005   

Price                                                                 \
Ticker            AMZN       GOOG      GOOGL        META        MSFT   
Date                                                                   
2010-01-04    6.830500  15.678981  15.753504         NaN   31.100000   
2010-01-05    6.774000  15.637387  15.711712         NaN   31.100000   
...                ...        ...        ...         ...         ...   
2020-12-29  167.532501  89.622002  89.423500  280.510010  227.179993   
2020-12-30  167.104996  88.254753  88.388000  278.079987  225.630005   

Price                          Low                                    \
Ticker            TSLA        AAPL        AMZN       GOOG      GOOGL   
Date                                                                   
2010-01-04         NaN    7.585000    6.657000  15.547723  15.621622   
2010-01-05         NaN    7.616071    6.590500  15.480475  15.554054   
...                ...         ...         ...        ...        ...   
2020-12-29  223.300003  134.339996  164.061005  87.804497  87.755501   
2020-12-30  232.199997  133.399994  164.123505  86.279999  86.400002   

Price                                                 Open              \
Ticker            META        MSFT        TSLA        AAPL        AMZN   
Date                                                                     
2010-01-04         NaN   30.590000         NaN    7.622500    6.812500   
2010-01-05         NaN   30.639999         NaN    7.664286    6.671500   
...                ...         ...         ...         ...         ...   
2020-12-29  276.279999  223.580002  218.333328  138.050003  165.496994   
2020-12-30  271.709991  221.470001  222.786667  135.580002  167.050003   

Price                                                                 \
Ticker           GOOG      GOOGL        META        MSFT        TSLA   
Date                                                                   
2010-01-04  15.615220  15.689439         NaN   30.620001         NaN   
2010-01-05  15.620949  15.695195         NaN   30.850000         NaN   
...               ...        ...         ...         ...         ...   
2020-12-29  89.389503  89.361504  276.950012  226.309998  220.333328   
20

## Calculate metrics

In [25]:
df = df_base['Adj Close']

df = (df
 .loc['2010':]
 .groupby(df.index.year).pct_change().add(1)
 .groupby(df.index.year).cumprod().sub(1)
 .resample('1Y').last()
 .mul(100)
)

/var/folders/xm/_8bv9n2n5f3f64xxt_d5_l880000gn/T/ipykernel_56999/646562998.py:5: FutureWarning: The default fill_method='ffill' in DataFrameGroupBy.pct_change is deprecated and will be removed in a future version. Call ffill before calling pct_change to retain current behavior and silence this warning.
  .groupby(df.index.year).pct_change().add(1)


In [ ]:
df.style

Ticker           AAPL       AMZN       GOOG      GOOGL       META       MSFT  \
Date                                                                           
2010-12-31  50.721936  34.428675  -5.230155  -5.230155        NaN  -7.944166   
2011-12-31  22.887383  -6.036268   6.875145   6.875157        NaN  -4.754562   
...               ...        ...        ...        ...        ...        ...   
2019-12-31  88.742471  20.057437  27.840510  26.994903  51.275050  58.259240   
2020-12-31  79.623539  73.120795  27.216482  26.855813  29.597678  39.477251   

Ticker            TSLA  
Date                    
2010-12-31   11.469194  
2011-12-31    7.287734  
...                ...  
2019-12-31   34.892938  
2020-12-31  707.395544  

[11 rows x 7 columns]

## Interpretation with AI

1. Copy table to clipboard
2. Create a prompt for ChatGPT in [this folder](../prompts/README.md)
3. Paste table into ChatGPT, along with the prompt
4. Paste the response below

Aquí están algunas explicaciones sobre los valores más significativos, tanto positivos como negativos, en los retornos acumulados anuales de los activos financieros, respaldados por noticias:

1. **Apple (AAPL) 2010-2020**:
   - **2010 (50.72%)**: Apple tuvo un excelente año debido al lanzamiento del iPhone 4 y el iPad original, ambos revolucionarios en sus respectivas categorías. Estos productos incrementaron significativamente las ventas y los ingresos de la compañía【7†source】.
   - **2014 (42.63%)**: En este año, Apple realizó una división de acciones 7:1 en junio, lo que hizo que sus acciones fueran más accesibles para los inversores. Además, el lanzamiento del iPhone 6 y 6 Plus impulsó las ventas a nuevos récords【8†source】【16†source】.
   - **2020 (79.62%)**: En medio de la pandemia de COVID-19, Apple lanzó el iPhone 12 con tecnología 5G y experimentó un aumento significativo en la demanda de sus productos y servicios debido al aumento del trabajo y estudio desde casa【15†source】【17†source】.

2. **Tesla (TSLA) 2020 (707.40%)**:
   - Tesla tuvo un crecimiento impresionante en 2020 impulsado por varios factores, incluyendo su inclusión en el índice S&P 500, lo cual atrajo una gran cantidad de inversiones. Además, reportó entregas récord de vehículos y continuó expandiéndose en nuevos mercados【16†source】.

3. **Amazon (AMZN) 2015 (119.07%)**:
   - Amazon experimentó un aumento significativo en 2015 gracias al rápido crecimiento de Amazon Web Services (AWS), que se convirtió en un importante motor de ingresos. El evento Prime Day también contribuyó al incremento en las ventas y al rendimiento positivo de las acciones【15†source】.

4. **Meta (META) 2013 (95.18%)**:
   - En 2013, Facebook (ahora Meta) tuvo un año excepcional debido a su salida a bolsa y su creciente popularidad y expansión global. La compañía también comenzó a monetizar su plataforma móvil de manera más efectiva, lo que impulsó los ingresos y el precio de sus acciones【15†source】.

5. **Microsoft (MSFT) 2019 (58.26%)**:
   - Microsoft tuvo un gran desempeño en 2019 gracias al crecimiento continuo de sus servicios en la nube, especialmente Azure, que se convirtió en un motor clave de ingresos. También se beneficiaron de la expansión de su suite Office 365 y la adquisición de GitHub【15†source】.

Estos eventos reflejan cómo los lanzamientos de productos innovadores, las estrategias empresariales y los cambios en el mercado pueden influir significativamente en los retornos de las acciones. Para más detalles, puedes consultar las siguientes fuentes:
- [Apple Historical Data](https://www.nasdaq.com)
- [Apple Stock News](https://stockanalysis.com)
- [Apple Stock Price History](https://markets.businessinsider.com)